packages/collector/src/uncaught/index.js

```diff
@@ -61,123 +56,16 @@ exports.init = function (_config, _downstreamConnection, _processIdentityProvide
   config = _config;
   downstreamConnection = _downstreamConnection;
   processIdentityProvider = _processIdentityProvider;
-  if (config.reportUncaughtException) {
-    if (!processIdentityProvider) {
-      logger.warn('Reporting uncaught exceptions is enabled, but no process identity provider is available.');
-    } else if (typeof processIdentityProvider.getEntityId !== 'function') {
-      logger.warn(
-        'Reporting uncaught exceptions is enabled, but the process identity provider does not support ' +
-          'retrieving an entity ID.'
-      );
-    }
-  }
 };
 
 exports.activate = function () {
-  activateUncaughtExceptionHandling();
   activateUnhandledPromiseRejectionHandling();
 };
 
-function activateUncaughtExceptionHandling() {
-  if (config.reportUncaughtException) {
-    process.once(uncaughtExceptionEventName, onUncaughtException);
-    logger.warn(
-      'Reporting uncaught exceptions is enabled. This feature is deprecated. Please consider disabling it ' +
-        'and rely on https://www.instana.com/docs/ecosystem/os-process/#abnormal-termination instead.'
-    );
-    if (process.version === 'v12.6.0') {
-      logger.warn(
-        'You are running Node.js v12.6.0 and have enabled reporting uncaught exceptions. To enable this feature, ' +
-          '@instana/collector will register an uncaughtException handler. Due to a bug in Node.js v12.6.0, the ' +
-          'original stack trace will get lost when this process is terminated with an uncaught exception. ' +
-          'Instana recommends to use a different Node.js version (<= v12.5.0 or >= v12.6.1). See ' +
-          'https://github.com/nodejs/node/issues/28550 for details.'
-      );
-    }
-  } else {
-    logger.info('Reporting uncaught exceptions is disabled.');
-  }
-}
-
 exports.deactivate = function () {
-  process.removeListener(uncaughtExceptionEventName, onUncaughtException);
```

Comment:
I was wondering why the feature got deprecated in 1.112. https://github.com/instana/nodejs/commit/5fda92a48032f06951578c3ac874d69cc089651f does not really explain it :)

I understand that with https://www.instana.com/docs/ecosystem/os-process/#abnormal-termination we can tell if an app has crashed. But I wonder how the customer will figure out the reason behind the crash? (if he doesn't use rollbar or similar)

In general we do report errors using `span.ec` and `span.error` when finishing spans and problems were detected.

Reply:
True, for the reason of the crash one would need to check the log files of the application or similar means. The two features are not fully equivalent, but the abnormal termination feature is much more general (does not work only for Node.js).

The reason it got deprecated is in part because of the following crux: To report any data from a process that is about to terminate you need **synchronous** HTTP communication, something that is simply not available in Node.js. The current work around (starting a child process and waiting for it to terminate) is not great.

In particular, it delays the process termination, which will also delay starting a new process when some form of orchestration (K8s) or process monitoring ([forever](https://www.npmjs.com/package/forever), [nodemon](https://nodemon.io/) etc.) is used. Thus, this feature prolongs the downtime of the service under monitoring.